go.mod

```diff
@@ -80,3 +80,7 @@ require (
 	google.golang.org/genproto/googleapis/api v0.0.0-20230913181813-007df8e322eb // indirect
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20230913181813-007df8e322eb // indirect
 )
+
+// The Buf CLI is the one library that should not depend on BSR generated SDKs
+// as this could cause bootstrapping errors.
+replace buf.build/gen/go/bufbuild/protovalidate/protocolbuffers/go/buf/validate => ./private/gen/proto/go/buf/validate
```

Comment:
Is this true? What bootstrapping errors do we expect? If we're using the `buf` CLI to work on `buf`, we're working with a binary where the remote packages are already resolved and compiled, so I don't see how this is a problem here.

I was under the impression that it's the core repo that must not depend on generated SDKs. Note that this directive here will do nothing to prevent a cyclic dependency in core since replace directives in dependencies are ignored by the go tool.

Reply:
>we're working with a binary where the remote packages are already resolved and compiled

The `buf` CLI's binary itself will be fine, since like you said, it's just a binary.
> Note that this directive here will do nothing to prevent a cyclic dependency in core since replace directives in dependencies are ignored by the go tool.

Yes I think this is the concern and we will also need to add a replace directive there, before or around the time when this PR is merged.

Reply:
> Yes I think this is the concern

So we can remove the replace directive here?

Reply:
Great question, and with the reasoning above I think it's probably fine? @bufdev what do you think?